### PR TITLE
Inform user of docker tag configuration

### LINKED ISSUE #444: Should inform user can change default Docker tag when building to Docker daemon.
To replace the default `<project name>:<project version>`.

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #444.

Most of the changes here are just adding tests.

### CODE REVIEW

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java [hunk 1]
```diff
@@ -142,4 +138,27 @@ BuildDockerTask setJibExtension(JibExtension jibExtension) {
...
+   * @return an {@link ImageReference} parsed from the configured target image, or one of the form
+   *     "project-name:project-version" if target image is not configured
+   */
+  ImageReference getDockerTag(GradleBuildLogger gradleBuildLogger)
+      throws InvalidImageReferenceException {
+    Preconditions.checkNotNull(jibExtension);
+    if (Strings.isNullOrEmpty(jibExtension.getTargetImage())) {
+      // TODO: Validate that project name and version are valid repository/tag
+      gradleBuildLogger.lifecycle(
+          "Using default docker tag "
```
[the remark this answers is not in the document]
Comment (coollog) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Yea, `"Tagging image with generated image reference xxx:yyyy.` would be even more accurate.

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java [hunk 2]
```diff
@@ -142,4 +138,27 @@ BuildDockerTask setJibExtension(JibExtension jibExtension) {
     this.jibExtension = jibExtension;
     return this;
   }
+
+  /**
+   * @param gradleBuildLogger the logger used to notify users of the target image parameter
+   * @return an {@link ImageReference} parsed from the configured target image, or one of the form
```
[the remark this answers is not in the document]
Comment (TadCordle) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Yeah, this is something that we haven't been following throughout the project. It seems like the javadoc checker doesn't error on missing descriptions (but maybe it should?)

And no worries, don't hesitate to be pedantic!

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java [hunk 3]
```diff
@@ -142,4 +138,30 @@ BuildDockerTask setJibExtension(JibExtension jibExtension) {
     this.jibExtension = jibExtension;
     return this;
   }
+
+  /**
+   * Returns an {@link ImageReference} retrieved from the build configuration, or the project name/
```
Comment (coollog) [inline jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Something like `or {@code <the project name>/<version>}` to make its format clear, or what you did in `@return`

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java [hunk 4]
```diff
@@ -142,4 +138,30 @@ BuildDockerTask setJibExtension(JibExtension jibExtension) {
...
+   * @return an {@link ImageReference} parsed from the configured target image, or one of the form
+   *     "project-name:project-version" if target image is not configured
+   */
+  ImageReference getDockerTag(GradleBuildLogger gradleBuildLogger)
+      throws InvalidImageReferenceException {
+    Preconditions.checkNotNull(jibExtension);
+    if (Strings.isNullOrEmpty(jibExtension.getTargetImage())) {
+      // TODO: Validate that project name and version are valid repository/tag
+      gradleBuildLogger.lifecycle(
+          "Tagging image with generated image reference "
```
Comment (coollog) [inline jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java]: Could add this to `HelpfulSuggestions` (Maven/Gradle specifics in `HelpfulSuggestionsProvider`), or have a TODO for now

**Pull request merged**